frontend/src/components/module-viewer/content/modals/EditHeadingModal.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  Select,
+  FormControl,
+  FormLabel,
+  Input,
+  Heading,
+} from "@chakra-ui/react";
+import React, { useContext, useState } from "react";
+import { HeadingBlockState } from "../../../../types/ContentBlockTypes";
+import { Modal } from "../../../common/Modal";
+import EditorContext from "../../../../contexts/ModuleEditorContext";
+import {
+  EditContentModalProps,
+  ValidHeadingSizes,
+} from "../../../../types/ModuleEditorTypes";
+
+const EditHeadingModal = ({
+  block,
+  index,
+  isOpen,
+  onClose,
+}: EditContentModalProps<HeadingBlockState>): React.ReactElement => {
+  const [text, setText] = useState(block.content.text ?? "why");
+  const [size, setSize] = useState(
+    block.content.size ?? ValidHeadingSizes.heading1,
+  );
+  const [invalid, setInvalid] = useState(false);
+  const context = useContext(EditorContext);
+  if (!context) return <></>;
+  const { dispatch } = context;
+
+  const onSave = () => {
+    if (!(size && text)) {
+      setInvalid(true);
+      return;
+    }
+    const newBlock = {
+      ...block,
+      content: {
+        ...block.content,
+        text,
+        size,
+      },
+    };
+
+    dispatch({
+      type: "update-block",
+      value: {
+        index,
+        block: newBlock,
+      },
+    });
+
+    onClose();
+  };
+
+  return (
+    <Modal
+      size="xl"
+      header="Heading"
+      onConfirm={onSave}
+      onCancel={onClose}
+      isOpen={isOpen}
+    >
+      <FormControl isInvalid={invalid} isRequired>
```

Comment:
As per my understanding, it's generally better to use one `FormControl` per input element. https://chakra-ui.com/docs/components/form-control